pkg/server/ingest.go

```diff
@@ -65,17 +65,24 @@ func (h ingestHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		if err == nil {
 			var profile *tree.Profile
 			var prevProfile *tree.Profile
+
 			f, _, err := r.FormFile("profile")
-			if err == nil {
-				profile, err = convert.ParsePprof(f)
+			if err != nil {
+				WriteError(h.log, w, http.StatusInternalServerError, err, "error happened while getting profile file")
+				return
+			}
+
+			profile, err = convert.ParsePprof(f)
+			if err != nil {
+				WriteError(h.log, w, http.StatusInternalServerError, err, "error happened while parsing profile file")
+				return
 			}
+
 			f, _, err = r.FormFile("prev_profile")
 			if err == nil {
-				prevProfile, err = convert.ParsePprof(f)
+				prevProfile, _ = convert.ParsePprof(f)
```

Comment:
I think in that case `err` shouldn't be nil in the previous line and we shouldn't try to parse it, right? I think that not having the previous profile is fine, but if it's there and the parsing fails, then it should be considered an error.

Reply:
@abeaumont do you suggest returning Internal Server Error on failed parsing `prev_error` too? I can do this.

By the way, in my opinion, this function handles errors in not so conventional manner. Do you think it may benefit from some refactoring, splitting into several functions? I can try to purpose some changes in separate PR, of course, if you guys think it will be appropriate.

Reply:
@abeaumont push new version, with 500 on failed parsing of `prev_profile`. Tried it locally and can't make it fire in a happy scenario.

Reply:
> @abeaumont do you suggest returning Internal Server Error on failed parsing prev_error too? I can do this.

Yes, exactly :+1: 

> Do you think it may benefit from some refactoring, splitting into several functions?

Yes, I agree this code block could be improved, so that would be very welcome!